### PR TITLE
Fixes for cmake shared library detection.

### PR DESCRIPTION
**Pull request by Zopolis4**
These changes were originally going to be part of a series of changes that let dolphin build on ChromeOS, but then it...built. Anyways, these should make the cmake build steps a bit less noisy, now that it isint complaining about not having a Findfmt.cmake and hidapi can see FindLIBUDEV.cmake. 

Most of the Find*.cmake files also need a bit of an overhaul, they're wildly inconsistent on capitalisation, indentation, ordering, naming, arguments, etc. I don't feel like I know enough about cmake or the coding styles, but I thought I'd put it out there. We should also have a FindQt.cmake, It'd remove the need for all the CMAKE_PREFIX_PATH nonsense and should make building on esoteric platforms or configurations much easier. Again, I don't know enough about cmake to confidently make one, just putting it out there.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -650,9 +650,7 @@ endif()
 add_subdirectory(Externals/cpp-optparse)
 
 find_package(fmt 8)
-if(fmt_FOUND)
-  message(STATUS "Using shared fmt ${fmt_VERSION}")
-else()
+if(NOT fmt_FOUND)
```
Comment (AdmiralCurtiss) [inline CMakeLists.txt:653]: Does `find_package(fmt 8 QUIET)` not work for this one?

Comment (Zopolis4) [reply CMakeLists.txt:653]: It does, but is very noisy if it doesn't find it, and this is more robust.

**Comment by Rumi-Larry** [comment]
Do ".cmake" files not need a legal header?

**Comment by Zopolis4** [comment]
> Do ".cmake" files not need a legal header?

A bunch of them don't, it's a bit of a mess. I'm not sure, a couple of them that we presumably sourced from elsewhere don't.

File: CMakeLists.txt (hunk 1 above)
Comment (Tilka) [reply CMakeLists.txt:653]: What do you mean by robust?

Comment (Zopolis4) [reply CMakeLists.txt:653]: Able to find more types of installations

**Comment by Zopolis4** [comment]
I've removed the `findfmt.cmake` due to conflicts with target importing.

**Review by Pokechu22 — APPROVED** [review]
Based on https://cmake.org/cmake/help/latest/command/find_package.html#search-modes it is case-sensitive for `Find<PackageName>.cmake` (implied by the existence of `<lowercasePackageName>-config.cmake` and `<PackageName>Config.cmake` below), and the variable name is `<PackageName>_FOUND` (for which we use `LIBUDEV_FOUND` and not `Libudev_FOUND`), so this seems reasonable to me. Untested.

**Pull request merged**